### ISSUE TITLE
An interesting issue related to the use_shell variable

### ISSUE THREAD

**Issue by howyoungchen** [bmaltais/kohya_ss#2361]
I encountered an error when using the verify_lora function of the lora tool. The error occurred due to the[ following code](https://github.com/bmaltais/kohya_ss/blob/80091ee701462e0a341ffbb693b9ee81f628d5fd/kohya_gui/verify_lora_gui.py#L53):
```
process = subprocess.Popen(
    run_cmd,
    stdout=subprocess.PIPE,
    stderr=subprocess.PIPE,
    env=env,
    shell=use_shell,
)
```

run_cmd is a string, but use_shell is set to False.

I first suspected the issue might be at [line 2336](https://github.com/bmaltais/kohya_ss/blob/80091ee701462e0a341ffbb693b9ee81f628d5fd/kohya_gui/lora_gui.py#L2336) in lora_gui.py:
```
with gr.Tab("Tools"):
    lora_tools = LoRATools(headless=headless)
```
    
Here, the use_shell_flag=use_shell_flag was missing.

However, after the modification, even though I made sure that use_shell was True [in the following function](https://github.com/bmaltais/kohya_ss/blob/80091ee701462e0a341ffbb693b9ee81f628d5fd/kohya_gui/verify_lora_gui.py#L141), verify_lora still showed use_shell as False when called, which was quite bizarre. Later, I changed visible=True, and all problems were resolved. I don’t understand what happened. It seems like a tricky bug.

```
verify_button.click(
    verify_lora,
    inputs=[
        lora_model,
        gr.Checkbox(value=use_shell, visible=False),
    ],
    outputs=[lora_model_verif_output, lora_model_verif_error],
    show_progress=False,
)
```

**Comment by bmaltais** [comment]
Thank you for raising this. Yes, this is strange… I will take a look at it today. Perhaps I will just do like I did for wf14 and just rewrite it to use a dictionary and enforce shell=False at all time…

Does WD14 caption work for you? On Linux or windows?

**Comment by bmaltais** [comment]
I have pushed a fix to the dev branch that does not require the use of shell anymore... simpler that way.

**Comment by howyoungchen** [comment]
@bmaltais 
Thank you for your prompt attention to the issue. My device runs Windows 11 with WSL2, and the operating system is Ubuntu. Indeed, I encountered a similar error previously while using WD14, but at that time, I simply circumvented the issue by forcefully setting the 'use shell' variable to true. However, this time the solution isn't as straightforward, which is why I'm puzzled about where the problem lies. A simple debug suggests that it seems to be caused by the internal mechanism of Gradio's click. I hope this information can be of help to you.
